src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1560,18 +1573,105 @@ void AdaptDispatch::_InitTabStopsForWidth(const size_t width)
 }
 
 //Routine Description:
-// Designate Charset - Sets the active charset to be the one mapped to wch.
-//     See DispatchTypes::VTCharacterSets for a list of supported charsets.
-//     Also http://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-Controls-beginning-with-ESC
-//       For a list of all charsets and their codes.
+// DOCS - Selects the coding system through which character sets are activated.
+//     When ISO2022 is selected, the code page is set to ISO-8859-1, and both
+//     GL and GR areas of the code table can be remapped. When UTF8 is selected,
+//     the code page is set to UTF-8, and only the GL area can be remapped.
+//Arguments:
+// - codingSystem - The coding system that will be selected.
+// Return value:
+// True if handled successfully. False otherwise.
+bool AdaptDispatch::DesignateCodingSystem(const wchar_t codingSystem)
+{
+    // If we haven't previously saved the initial code page, do so now.
+    // This will be used to restore the code page in response to a reset.
+    if (_initialCodePage == 0)
+    {
+        _pConApi->GetConsoleOutputCP(_initialCodePage);
+    }
```

Comment:
I know apps shouldn't mix these things, but what happens if we DCS, then SetConsoleOutputCP() through the API, then reset?

Reply:
It'll reset to whatever was saved in `_initialCodePage`. Ideally that would be the code page at startup, because the whole point of `RIS` is to reset to the start state. So for a cmd.exe shell you'd want something like CP437, while a bash shell would typically be UTF8. Unfortunately there's no easy way to determine that startup code page, since it's not yet set at the time the `AdapDispatch` is constructed. So using the value at the time of the first DOCS call seemed like a reasonable compromise to me. 90% of the time it's probably what you want. And if it isn't, it's not the end of the world.

That said, at some point we might want to consider adding a mechanism whereby the `AdaptDispatch` class could register a callback to be notified of console state changes (like the code page). And that should enable us to do something a little better here. I don't think it's essential for now though.